classes/curlmanager_security_helper.php

```diff
@@ -67,15 +60,23 @@ private function log_curl_http_requests($urlstring) : bool {
             $host = $url->get_host();
         } catch (\moodle_exception $e) {
             // Moodle exception is thrown if the $urlstring is invalid.
-            return false;
+            return true;
         }
 
+        // Check if the host is in allowed list.
+        $returnvalue = $this->host_is_allowed($host);
+
+        // Call moodle curl_security_helper method url_is_blocked.
+        $curlsecurityhelper = new curl_security_helper();
+        $urlblocked = $curlsecurityhelper->url_is_blocked($urlstring);
+
         $codepath = '';
         $rootcodepath = '';
         $trace = debug_backtrace();
-        if (isset($trace[3]['file'])) {
-            $rootcodepath = $trace[3]['file'];
-            $codepath = str_replace('/siteroot', '', $trace[3]['file']);
+        $lasttrace = count($trace) - 1;
+        if (isset($trace[$lasttrace]['file'])) {
+            $rootcodepath = $trace[$lasttrace]['file'];
+            $codepath = str_replace('/siteroot', '', $rootcodepath);
```

Comment:
what is /siteroot here? this looks like something hardcoded and should be $CFG->dirroot maybe?